conda/common/url.py

```diff
@@ -22,9 +24,15 @@
 from .compat import on_win
 from .path import split_filename, strip_pkg_extension
 
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+    from re import Pattern
+    from typing import Any, Self
+    from urllib.parse import ParseResult
+
 
-def hex_octal_to_int(ho):
-    ho = ord(ho.upper())
+def hex_octal_to_int(ho: str) -> int:
```

Comment:
Should this be `return int(ho, 16)`

Reply:
The rationale seems to be "more control". [See this comment](https://github.com/conda/conda/commit/eead59d1810c685089ccdbe36bbd000f2cfa1931#diff-b065cdb7cde0388abd733125ec99e60f558f08ebd44d4d36823cdd5ec69378a5R96-R105).

Reply:
Opened https://github.com/conda/conda/pull/14840